MIOpen generator fails

When trying to generate wrapper for MIOpen, it fails with the error below.
It does produce wrappers for some parts of the library before it fails though.
My guess, is that it may be due to following anonymous structs: [link](https://github.com/ROCmSoftwarePlatform/MIOpen/blob/84f1ce130fc1c5df8fa9d5a4cb7edf62b978d0fe/include/miopen/miopen.h#L1111).

Code for the generation: [link](https://github.com/pxl-th/AMDGPU.jl/blob/pxl-th/miopen/src/miopen/generator/miopen-generator.jl).

**Error:**

```julia
pxl-th@Yotun:~/.julia/dev/AMDGPU/src/miopen/generator$ jlt miopen-generator.jl 
[ Info: Parsing headers...
/home/pxl-th/.julia/artifacts/47f9f7e8df1e973a09db0937199b47504c8081db/include/miopen/config.h:126:10: fatal error: 'hip/hip_version.h' file not found
/home/pxl-th/.julia/artifacts/47f9f7e8df1e973a09db0937199b47504c8081db/include/miopen/config.h:126:10: fatal error: 'hip/hip_version.h' file not found
[ Info: Processing header: /home/pxl-th/.julia/artifacts/47f9f7e8df1e973a09db0937199b47504c8081db/include/miopen/config.h
[ Info: Processing header: /home/pxl-th/.julia/artifacts/47f9f7e8df1e973a09db0937199b47504c8081db/include/miopen/export.h
[ Info: Processing header: /home/pxl-th/.julia/artifacts/47f9f7e8df1e973a09db0937199b47504c8081db/include/miopen/miopen.h
[ Info: Processing header: /home/pxl-th/.julia/artifacts/47f9f7e8df1e973a09db0937199b47504c8081db/include/miopen/version.h
[ Info: Building the DAG...
[ Info: Emit Julia expressions...
[ Info: [ProloguePrinter]: print to ./libMIOpen.jl
[ Info: [GeneralPrinter]: print to ./libMIOpen.jl
ERROR: LoadError: AssertionError: length(child_nodes) == length(fields)
Stacktrace:
 [1] pretty_print(io::IOStream, node::ExprNode{Clang.Generators.StructAnonymous, Clang.CLStructDecl}, options::Dict{String, Any})
   @ Clang.Generators ~/.julia/packages/Clang/yiVXf/src/generator/print.jl:119
 [2] (::Clang.Generators.var"#135#136"{ExprDAG, Dict{String, Any}, Nothing, Vector{Any}, Dict{String, Any}})(io::IOStream)
   @ Clang.Generators ~/.julia/packages/Clang/yiVXf/src/generator/passes.jl:955
 [3] open(::Clang.Generators.var"#135#136"{ExprDAG, Dict{String, Any}, Nothing, Vector{Any}, Dict{String, Any}}, ::String, ::Vararg{String}; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ Base ./io.jl:395
 [4] open(::Function, ::String, ::String)
   @ Base ./io.jl:392
 [5] (::GeneralPrinter)(dag::ExprDAG, options::Dict{String, Any})
   @ Clang.Generators ~/.julia/packages/Clang/yiVXf/src/generator/passes.jl:951
 [6] build!(ctx::Context, stage::Clang.Generators.BuildStage)
   @ Clang.Generators ~/.julia/packages/Clang/yiVXf/src/generator/context.jl:169
 [7] build!(ctx::Context)
   @ Clang.Generators ~/.julia/packages/Clang/yiVXf/src/generator/context.jl:160
 [8] top-level scope
   @ ~/.julia/dev/AMDGPU/src/miopen/generator/miopen-generator.jl:19
in expression starting at /home/pxl-th/.julia/dev/AMDGPU/src/miopen/generator/miopen-generator.jl:19
```

--- a/src/generator/preprocessing.jl
+++ b/src/generator/preprocessing.jl
@@ -141,7 +141,7 @@ function nested_anonymous_check end
 
 nested_anonymous_check(dag::ExprDAG, node::ExprNode) = false
 
-function nested_anonymous_check(dag::ExprDAG, node::ExprNode{<:Union{StructDefinition,StructMutualRef,UnionDefinition}})
+function nested_anonymous_check(dag::ExprDAG, node::ExprNode{<:Union{StructAnonymous,StructDefinition,StructMutualRef,UnionDefinition}})
     for c in fields(getCursorType(node.cursor))
         ty = getCursorType(c)
         if occursin(__ANONYMOUS_MARKER, spelling(ty)) || occursin(__UNNAMED_MARKER, spelling(ty))

--- a/test/generators.jl
+++ b/test/generators.jl
@@ -133,6 +133,6 @@ end
 end
 
 @testset "Issue 233" begin
-    ctx = create_context(joinpath(@__DIR__, "include/test_u.h"), get_default_args())
+    ctx = create_context(joinpath(@__DIR__, "include/union-in-anon-struct.h"), get_default_args())
     @test build!(ctx) isa Any
 end

--- a/test/include/union-in-anon-struct.h
+++ b/test/include/union-in-anon-struct.h
@@ -1,9 +1,9 @@
 // Issue 233
-struct Meow {
+typedef struct {
     int blah;
     union {
         float either1;
         int either2;
     };
     float moo;
-};
+} Meow;

